foxtrot-server/src/main/java/com/flipkart/foxtrot/server/di/FoxtrotModule.java

```diff
@@ -0,0 +1,209 @@
+               ? new EsIndexOptimizationConfig()
+               : configuration.getEsIndexOptimizationConfig();
+    }
+
+    @Provides
+    @Singleton
+    public DataDeletionManagerConfig dataDeletionManagerConfig(FoxtrotServerConfiguration configuration) {
+        return configuration.getDeletionManagerConfig();
+    }
+
+    @Provides
+    @Singleton
+    public ConsoleHistoryConfig consoleHistoryConfig(FoxtrotServerConfiguration configuration) {
+        return null == configuration.getConsoleHistoryConfig()
+               ? new ConsoleHistoryConfig()
+               : configuration.getConsoleHistoryConfig();
+    }
+
+    @Provides
+    @Singleton
+    public CacheConfig cacheConfig(FoxtrotServerConfiguration configuration) {
+        return configuration.getCacheConfig();
+    }
+
+    @Provides
+    @Singleton
+    public EmailConfig emailConfig(FoxtrotServerConfiguration configuration) {
+        return configuration.getEmailConfig();
+    }
+
+    @Provides
+    @Singleton
+    public SegregationConfiguration segregationConfiguration(FoxtrotServerConfiguration configuration) {
+        return configuration.getSegregationConfiguration();
+    }
+
+    @Provides
+    @Singleton
+    public ObjectMapper objectMapper(Environment environment) {
+        return environment.getObjectMapper();
+    }
+
+    @Provides
+    @Singleton
+    public List<IndexerEventMutator> provideMutators(
+            FoxtrotServerConfiguration configuration,
+            ObjectMapper objectMapper) {
+        return Collections.singletonList(new LargeTextNodeRemover(objectMapper, configuration.getTextNodeRemover()));
+    }
+
+    @Provides
+    @Singleton
+    public List<ActionExecutionObserver> actionExecutionObservers(CacheManager cacheManager, InternalEventBus eventBus) {
+        return ImmutableList.<ActionExecutionObserver>builder()
+                .add(new MetricRecorder())
+                .add(new ResponseCacheUpdater(cacheManager))
+                .add(new SlowQueryReporter())
+                .add(new EventPublisherActionExecutionObserver(eventBus))
+                .build();
+    }
+
+    @Provides
+    @Singleton
+    public ExecutorService provideGlobalExecutorService(Environment environment) {
+        return environment.lifecycle()
+                .executorService("query-executor-%s")
+                .minThreads(20)
+                .maxThreads(30)
+                .keepAliveTime(Duration.seconds(30))
+                .build();
+    }
+
+    @Provides
+    @Singleton
+    public ScheduledExecutorService provideGlobalScheduledExecutorService(Environment environment) {
+        return environment.lifecycle()
+                .scheduledExecutorService("cardinality-executor")
+                .threads(1)
+                .build();
+    }
+
+    @Provides
+    @Singleton
+    public List<HealthCheck> provideHealthChecks(Environment environment) {
+        return Collections.emptyList(); //TODO::REturn dropwizard healthchecks
```

Comment:
???

Reply:
implemented hbase and hazelcast healthchecks and added provider for `List<Healthcheck>`